Update docs with the new remediationStrategy spec (#145)
medik8s/fence-agents-remediation: README.md

  name: fenceagentsremediationtemplate-default
  namespace: default
spec:
  template:
    spec:
        remediationStrategy: <remediation_strategy>
```

> *Note*: FenceAgentsRemediationTemplate CR must be created in the same namespace that the FAR operator has been installed.
 

> The `.spec.template.spec.remediation_strategy` field can either be `ResourceDeletion` or `OutOfServiceTaint`:

Reviewer: It can be the `.spec.template.spec.remediation_strategy` field of fartemplate CR or simply `.spec.remediation_strategy` of FAR CR. I prefer the latter and modify the README in the [FAR CR fields](https://github.com/medik8s/fence-agents-remediation/pull/145/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R197-R202) rather than the template spec of fartemplate

> - `ResourceDeletion`: This remediation strategy removes the pods on the node, rather than the removal of the node object. This strategy recovers workloads faster.

Reviewer: Not sure if the rest is needed.
```suggestion
- `ResourceDeletion`: This remediation strategy (forcefully) removes the pods on the node.
```

> The `.spec.remediation_strategy` field defines the _remediationStrategy_ that FAR supports and will be used for recovering workloads faster during node reboot in a [non-graceful](https://kubernetes.io/docs/concepts/cluster-administration/node-shutdown/#non-graceful-node-shutdown) manner. There are only two possible remediations - `ResourceDeletion` or `OutOfServiceTaint`:

Author: You're right, not sure what happened. I'll fix it

> - `OutOfServiceTaint`: This remediation strategy implicitly causes the removal of the pods and associated volume attachments on the node, rather than the removal of the node object. It achieves this by placing the `OutOfServiceTaint` taint on the node. The `OutOfServiceTaint` strategy also represents a non-graceful node shutdown. A non-graceful node shutdown occurs when a node is shutdown and not detected, instead of triggering an in-operating system shutdown.

Reviewer: >The `OutOfServiceTaint` strategy also represents a non-graceful node shutdown. A non-graceful node shutdown occurs when a node is shutdown and not detected, instead of triggering an in-operating system shutdown.

I am not sure the above sentence is needed as it was already mentioned in line 162.

```suggestion
- `OutOfServiceTaint`: This remediation strategy implicitly causes the removal of the pods and associated volume attachments on the node. It achieves this by placing the [`OutOfServiceTaint` taint](https://kubernetes.io/docs/reference/labels-annotations-taints/#node-kubernetes-io-out-of-service) on the node.
```

* `retrycount` - number of times to retry the fence agent in case of failure. The default is 5.
* `retryinterval` - interval between retries in seconds. The default is "5s".
* `timeout` - timeout for the fence agent in seconds. The default is "60s".
* `remediationStrategy` - either `ResourceDeletion` or `OutOfServiceTaint`:

>     * `ResourceDeletion`: This remediation strategy removes the pods on the node.

Reviewer 2: Maybe it's me only, but why don't call things how they are called already? 🤔 
Here: we do not `remove` anything, we `delete` them... that's the actual method name we use. That's the `D` in CRUD. 🤷🏼‍♂️ (Same below)

(and I don't block on this, just a general thought, which I had more often at various places)

Reviewer 2: /lgtm

in case you want to change...
/hold

Author: > why don't we call things how they are called already

for me, I think it's just an automatically mechanism to avoid repetitions in written documents, but indeed when explaining something this habit can be limited :)

      worker-0: "6233"
      worker-1: "6234"
      worker-2: "6235"

>   remediationStrategy: OutOfServiceTaint

Reviewer: Although `OutOfServiceTaint` is preferable when it is enabled, but by default the `remediationStrategy` will be `ResourceDeletion` (and it is by default because due to issue of support to the _OutOfService_ taint)
